src/actor/npc/ac_npc_shop_master.c

```diff
@@ -0,0 +1,2677 @@
+    f32 z = 260.0;
+    if (((ACTOR *)shop_master)->world.position.z > z) {
+        ((ACTOR *)shop_master)->world.position.z = z;
+        aNSC_set_stop_spd(shop_master);
+        CLIP(npc_clip)->animation_init_proc((ACTOR *)shop_master, 0x5, 0x1);
+    }
+    if (mMsg_Check_MainNormalContinue(mMsg_Get_base_window_p()) == TRUE) {
+        int action = 0x4;
+        switch (mChoice_Get_ChoseNum(mChoice_Get_base_window_p())) {
+        case mChoice_CHOICE0:
+            action = 0x4;
+            break;
+        case mChoice_CHOICE1:
+            aNSC_set_make_basement_info();
+            action = 0x2;
+            break;
+        }
+        aNSC_setupAction(shop_master, game, action);
+    }
+}
+
+static void aNSC_check_roof_col_order(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game) {
+    f32 z = 260.0;
+    int unk = mDemo_Get_OrderValue(mDemo_ORDER_NPC0, 9);
+    if (((ACTOR *)shop_master)->world.position.z > z) {
+        ((ACTOR *)shop_master)->world.position.z = z;
+        aNSC_set_stop_spd(shop_master);
+        CLIP(npc_clip)->animation_init_proc((ACTOR *)shop_master, 0x5, 0x1);
+    } 
+    if (unk) {
+        if (mMsg_Check_MainNormalContinue(mMsg_Get_base_window_p()) == TRUE) {
+            mHm_hs_c *home = &Save_Get(homes)[mHS_get_arrange_idx(Common_Get(player_no))];
+            int action;
+            int choice = mChoice_Get_ChoseNum(mChoice_Get_base_window_p());
+            switch (choice) {
+            case 0:
+            case 1:
+            case 2:
+                action = 0x2;
+                home->size_info.next_size += 1;
+                home->ordered_outlook_pal = choice;
+                aNSC_set_rehouse_order_date(home);
+                break;
+            default:
+                action = 0x5;
+                break;
+            }
+            mDemo_Set_OrderValue(mDemo_ORDER_NPC0, 9, 0x0);
+            aNSC_setupAction(shop_master, game, action);
+        }
+    }
+}
+
+static void aNSC_check_roof_col_order2(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game) {
+    if (mMsg_Check_MainNormalContinue(mMsg_Get_base_window_p()) == TRUE) {
+        mHm_hs_c *home = &Save_Get(homes)[mHS_get_arrange_idx(Common_Get(player_no))];
+        int choice = mChoice_Get_ChoseNum(mChoice_Get_base_window_p());
+        int unk = mDemo_Get_OrderValue(mDemo_ORDER_NPC0, 9);
+        int action = 0;
+        switch (choice) {
+        case 0:
+        case 1:
+        case 2:
+            action = 0x2;
+            home->size_info.next_size += 1;
+            home->ordered_outlook_pal = choice + unk * 3;
+            aNSC_set_rehouse_order_date(home);
+            break;
+        default:
+            action = 0x5;
+        }
+        mDemo_Set_OrderValue(mDemo_ORDER_NPC0, 9, 0x0);
+        aNSC_setupAction(shop_master, game, action);
+    }
+}
+
+static void aNSC_present_balloon_start_wait(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game) {
+    if ((int)mDemo_Get_OrderValue(0x4, 0x1) == 2) {
```

Comment:
Firstly, if you have to cast like this I'd recommend moving it into an explicit variable. Secondly, 0x4 is the demo type, so it'd be `mDemo_TYPE_4` here.